IO/src/main/java/io/deephaven/io/streams/ByteBufferInputStream.java

```diff
@@ -341,48 +332,4 @@ public String readUTF() throws IOException {
 
         return new String(chars, 0, length);
     }
-
-    public String readUTF(@NotNull final CharSequenceAdapterBuilder output,
```

Comment:
If we're going to remove all of this, we need to consider whether we want to remove `CharSequenceAdapterBuilder`, and more realistically the entire `StringAlike` and `StringCache` framework.

Is this code hurting your use-case?

Reply:
It does look like all of that is dead code at this time, and we haven't had a use case for it in the 3.5ish years since setting up the fork. 

I am not sure I specifically need this PR at all - but as I was getting #188 to use this ByteBufferInputStream I noticed the commented out code, unusable code, and unused code that this PR removes. I would agree that CharSequenceAdapterBuilder should go too if we don't plan on using this (can always restore it later), and probably the rest of string.cache and friends likewise.

Reply:
Sorry, I do need to prune something here if this class will be usable in the client - AbstractCompressedString requires java.util.regex (which is subtly incompatible with js regex, so not emulated at all rather than confuse users).
ByteBufferInputStream -> CharSequenceAdapterBuilder -> MappedCompressedString -> AbstractCompressedString

We could extract a subtype of BBIS that retains its non-InputStream methods like `readUTF(CharSequenceAdapterBuilder)`, as another alternative to removing the methods?

Reply:
File a ticket for me to remove the residual compressed string complex of code, and let's merge this PR as-is.